lib/prepareFunctionsUpload.js

```diff
@@ -28,11 +28,28 @@ var _prepareSource = function(context, options) {
     logger.debug(err);
     throw new FirebaseError('Problem preparing functions directory for upload.', {exit: 1});
   }
-  return functionsConfig.materializeAll(getProjectId(options)).then(function(output) {
+
+  var next;
```

Comment:
How about just make this `var next = RSVP.resolve();` and then you can get rid of the `else`.